compiler/src/typed/typemod.re

```diff
@@ -908,75 +868,68 @@ let type_module = (~toplevel=false, funct_body, anchor, env, sstr /*scope*/) =>
 
   let signatures = List.map(resolve_type_alias, signatures);
 
-  let run = () => {
-    // TODO: Is it really safe to drop the import statements here?
-    let stritems = (statements, final_env);
-    (stritems, signatures, final_env);
-  };
-
-  run();
+  (statements, signatures, final_env);
 };
 
-let type_module = type_module(false, None);
+let type_module = type_module(None);
+
+let register_implicit_modules = modules => {
+  List.iter(
+    m => {
+      let filepath =
+        switch (m) {
+        | Grain_utils.Config.Pervasives_mod => "pervasives"
+        | Grain_utils.Config.Gc_mod => "runtime/gc"
+        };
+      Env.add_import(filepath);
+    },
+    modules,
+  );
+};
 
 let lookup_implicit_module_spec = m =>
   switch (m) {
-  | Grain_utils.Config.Pervasives_mod => ("Pervasives", "pervasives", true)
-  | Grain_utils.Config.Gc_mod => ("GC", "runtime/gc", false)
+  | Grain_utils.Config.Pervasives_mod => Some(("Pervasives", "pervasives"))
+  | Grain_utils.Config.Gc_mod => None
```

Comment:
The old boolean that was a part of this before indicated whether this module was added to the scope of the module. This was important before to make sure that GC got compiled, but since we changed how that happens this no longer needs to attempt to load/compile it. So the `None` here just means "nothing needs to be done for this."